feat: Challenge 39 based on filename as encryption key (#1023)
OWASP/wrongsecrets: src/main/resources/explanations/challenge39_hint.adoc

> This challenge can be solved by decrypting the base64 encoded secret in `secrchallenge.md`. You can do this either by
> 
> 1. Using an online aes decryption tool like https://www.devglan.com/online-tools/aes-encryption-decryption[https://www.devglan.com/online-tools/aes-encryption-decryption]
> - Copy the contents of the `secrchallenge.md` file and paste it in the textbox of the decryptor.
> - Make sure that the input format is `Base64` and cipher mode is `ECB`.
> - Use `secrchallenge.md` as decryption key and click on `Decrypt` to get the secret.
> 
> 2. Using the terminal
> - Launch the terminal while you are in the `executables` directory.
> - Type in `echo -n "secrchallenge.md" | xxd -p` to convert the plaintext key to hexadecimal key.
> - Then use the obtained decryption key to decrypt the file by typing `openssl enc -a -d -aes-128-ecb -in secrchallenge.md -K 736563726368616c6c656e67652e6d64 -out decrypted.md`
> - Copy the secret from the `decrypted.md` file in the `executables` folder.

Reviewer: ```suggestion
This challenge can be solved by decrypting the base64 encoded secret in `secrchallenge.md`. You can do this either by:

1. Using an online aes decryption tool like https://www.devglan.com/online-tools/aes-encryption-decryption[https://www.devglan.com/online-tools/aes-encryption-decryption]
- Copy the contents of the `secrchallenge.md` file and paste it into the textbox of the decryptor.
- Ensure the input format is `Base64` and the cipher mode is `ECB`.
- Use `secrchallenge.md` as decryption key and click `Decrypt` to get the secret.

2. Using the terminal
- Launch the terminal while you are in the `executables` directory.
- Type in `echo -n "secrchallenge.md" | xxd -p` to convert the plaintext key to a hexadecimal key.
- Then, use the obtained decryption key to decrypt the file by typing `openssl enc -a -d -aes-128-ecb -in secrchallenge.md -K 736563726368616c6c656e67652e6d64 -out decrypted.md`
- Copy the secret from the `decrypted.md` file in the `executables` folder.
```

This challenge can be solved by decrypting the base64 encoded secret in `secrchallenge.md`.

1. Copy the contents of the `secrchallenge.md` file.

> 2. Decode the copied string from base64 to UTF-8.

Reviewer 2: can you give examples how to do that in your terminal and/or online pages?

> 3. Decrypt it using any AES decryption tool with `secrchallenge.md` as decryption key.

Reviewer 2: Can you give an example how to do this with terminal or online?
